tools/contrib/fsurdat_modifier.py

```diff
@@ -0,0 +1,356 @@
+#! /usr/bin/env python
+"""
+|------------------------------------------------------------------|
+|---------------------  Instructions  -----------------------------|
+|------------------------------------------------------------------|
+
+Instructions for running on Cheyenne/Casper:
+
+load the following into your local environment
+    module load python
+    ncar_pylib
+
+-------------------------------------------------------------------
+To see available options:
+    ./fsurdat_modifier.py --help
+-------------------------------------------------------------------
+
+This script reads a surface dataset (fsurdat file) and outputs a
+modified copy of the same file.
+
+-------------------------------------------------------------------
+To run the script:
+    ./fsurdat_modifier.py
+ 
+To remove NPL (ncar_pylib) from your environment on Cheyenne/Casper:
+    deactivate
+-------------------------------------------------------------------
+
+"""
+
+#  Import libraries
+from __future__ import print_function
+
+import sys
+import os
+import string
+import logging
+import subprocess
+import argparse
+
+import xarray as xr
+
+from datetime import date
+from getpass import getuser
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+myname = getuser()
```

Comment:
- [ ] It seems non-standard to call a function to set a variable at the top level like this – and it won't work smoothly once this is broken up into pieces in the python directory. So this should just be inlined where it's needed.